### DISCUSSION TITLE
How to record streams as HLS files and prevent them from getting overwritten in case of any network disruptions?

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#4875] [FAQs] [answered]
I have a use case where a streamId will be reused but when I republish it overwrites previously recorded HLS files.
Even if I enable timestamp for recording it only adds the timestamp to MP4 files but not to .ts or m3u8 files. This issue becomes more critical when there is some disruption in the network that results in stopping and starting the stream in a short duration.

**Comment by yashtandon113** [comment, marked as answer]
Thank you for your question. Let me answer this question for two scenarios.

**1. Add new .ts files to previously recorded files to have a single recording.**
The.ts files with the same streamId can be appended to the same .m3u8 file. To do so, edit and add the following hlsflag setting to the red5-web.properties file under _/usr/local/antmedia/webapps/app-name/WEB-INF/ directory.

`settings.hlsflags=+append_list
`

For HLS recording, make sure that below properties are also added.
```
settings.hlsPlayListType=event
settings.deleteHLSFilesOnEnded=false
```
Then, restart the AMS with `sudo service antmedia restart`

**2. Record the stream with same streamId but without appending to old files.**
If you don't want the ts files to be appended to the previous recording, you may also enable date and timestamp for HLS files in versions 2.5.3 and higher by adding the following property.

`settings.addDateTimeToHlsFileName=true`

The result will be as follows:
![image](https://user-images.githubusercontent.com/82374739/223510382-04173252-1c57-4483-9a8d-22266f8a04d2.png)

https://antmedia.io/docs/guides/playing-live-stream/hls-playing/#save-hls-records

**Comment by Mohit-3196** [comment]
Hi All,
Let me give an update on this starting from Ant Media Sever v2.6.2, the application settings be it in Standalone Mode or Cluster mode can be changed directly from the Web Panel without having to edit the application configuration file `red5-web.properties` and restarting the server.

For more details check the application configuration document https://antmedia.io/docs/guides/configuration-and-testing/ams-application-configuration/

Thank you.

**Comment by sweetgoomba** [comment]
Hi @Mohit-3196 @yashtandon113 ,
Did you record the live streams as HLS format? I want to do the same thing but I don't know how.. Can you tell me how to achieve it?
According to Ant Media docs ([recording-live-streams](https://antmedia.io/docs/guides/recording-live-streams/)), it shows me only to record in mp4 or webM but I want the recorded files to be m3u8 and ts.
Thank you :)

**Reply by sweetgoomba** [reply]
Oh, i searched more and I found out [this](https://github.com/orgs/ant-media/discussions/6083) page and it leads me to [this docs](https://antmedia.io/docs/guides/playing-live-stream/hls-playing/#save-hls-records). I will try this! Thank you :-)

**Reply by yashtandon113** [reply]
@sweetgoomba 
Sure, please test and let us know if you have any questions.
